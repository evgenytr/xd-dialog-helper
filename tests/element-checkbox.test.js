const checkbox = require('../dialog-helper').types.CHECKBOX;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

const rendered = checkbox.render('dialog', {
    id: 'hr',
    label: 'Hello World'
}, actions);

describe('Checkbox Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();
        });
    });
    describe('value()', () => {
        it('should correctly return false as default value', () => {
            expect(checkbox.value(rendered)).toBe(false);
        });

        it('should correctly return true after the checkbox was checked', () => {
            rendered.input.click();
            expect(checkbox.value(rendered)).toBe(true);
        });
    });
    describe('validate()', () => {
        it('should always return true if there\'s no required prop', () => {
            expect(checkbox.valid(rendered)).toBeTruthy();
        });

        it('should return false for checked checkboxes if required == true', () => {
            let emptyField = checkbox.render('dialog',  {
                id: 'helloworld',
                label: 'My world',
                value: false,
                required: true
            }, actions);
            expect(checkbox.valid(emptyField)).toBeFalsy();
        });

        it('should return true for checked checkboxes if required == true', () => {
            let emptyField = checkbox.render('dialog',  {
                id: 'helloworld',
                label: 'My world',
                value: true,
                required: true
            }, actions);
            expect(checkbox.valid(emptyField)).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(checkbox.type).toMatchSnapshot();
    });
});
