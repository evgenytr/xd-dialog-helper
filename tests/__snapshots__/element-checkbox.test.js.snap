// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Input"`;

exports[`Checkbox Element render() should render correctly 1`] = `
Object {
  "input": <input
    id="dialog-hr"
    placeholder="Hello World"
    someattr="hello"
    type="checkbox"
  />,
  "props": Object {
    "htmlAttributes": Object {
      "someAttr": "hello",
    },
    "id": "hr",
    "label": "Hello World",
  },
  "type": Object {
    "render": [Function],
    "type": "Input",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-hr-wrapper"
    style="flex-direction: row; align-items: center;"
  >
    <input
      id="dialog-hr"
      placeholder="Hello World"
      someattr="hello"
      type="checkbox"
    />
    <span
      id="dialog-hr-label"
    >
      Hello World
    </span>
  </label>,
}
`;
